tests/api/api/test_functions.py

```diff
@@ -1000,19 +1002,43 @@ def failing_func():
         assert background_task.status.state == expected_status_result
 
 
+def test_list_functions_by_foo_spec(
+    db: sqlalchemy.orm.Session, client: fastapi.testclient.TestClient, monkeypatch
+):
+    project = "test-adam"
+    foo_spec = "adam"
+    function_1 = _generate_function(
+        function_name="adam-func", project=project, foo_spec=foo_spec, kind="job"
+    )
+    SQLDB().store_function(
```

Comment:
use either the fastapi client or the `get_db()` method for conventions consistency